src/Game/Jobs/RPR.ts

```diff
@@ -0,0 +1,1141 @@
+makeRPRResource(ResourceType.VoidShroud, 5); // Impossible for it to last 30s, but 30s is an upper bound
+makeRPRResource(ResourceType.Oblatio, 1, { timeout: 30 });
+makeRPRResource(ResourceType.EnhancedVoidReaping, 1, { timeout: 30 });
+makeRPRResource(ResourceType.EnhancedCrossReaping, 1, { timeout: 30 });
+
+makeRPRResource(ResourceType.IdealHost, 1, { timeout: 30 });
+makeRPRResource(ResourceType.PerfectioOcculta, 1, { timeout: 30 });
+makeRPRResource(ResourceType.PerfectioParata, 1, { timeout: 30 });
+
+makeRPRResource(ResourceType.ArcaneCircle, 1, { timeout: 20 }); // 20.00s exactly
+makeRPRResource(ResourceType.CircleOfSacrifice, 1, { timeout: 5 });
+makeRPRResource(ResourceType.BloodsownCircle, 1, { timeout: 6 });
+makeRPRResource(ResourceType.ImmortalSacrifice, 8, { timeout: 30 });
+
+makeRPRResource(ResourceType.ArcaneCrest, 1, { timeout: 5 });
+makeRPRResource(ResourceType.CrestOfTimeBorrowed, 1, { timeout: 5 });
+makeRPRResource(ResourceType.CrestOfTimeReturned, 1, { timeout: 15 });
+
+makeRPRResource(ResourceType.Soulsow, 1);
+makeRPRResource(ResourceType.Threshold, 1, { timeout: 10 });
+makeRPRResource(ResourceType.EnhancedHarpe, 1, { timeout: 10 });
+
+makeRPRResource(ResourceType.RPRCombo, 2, { timeout: 30 });
+makeRPRResource(RPRResourceType.RPRAoECombo, 1, { timeout: 30 });
+
+makeRPRResource(ResourceType.Feint, 1, {timeout: 15});
+makeRPRResource(ResourceType.TrueNorth, 1, {timeout: 10});
+makeRPRResource(ResourceType.ArmsLength, 1, {timeout: 6});
+makeRPRResource(ResourceType.Bloodbath, 1, {timeout: 20});
+
+export class RPRState extends GameState {
+    constructor(config: GameConfig) {
+        super(config);
+
+        const soulSliceStacks = Traits.hasUnlocked(TraitName.TemperedSoul, config.level) ? 2 : 1;
+        this.cooldowns.set(new CoolDown(ResourceType.cd_SoulSlice, 30, soulSliceStacks, soulSliceStacks));
+
+        this.cooldowns.set(new CoolDown(ResourceType.cd_ArcaneCircle, 120, 1, 1));
+
+        this.registerRecurringEvents();
+    }
+
+    refreshDeathsDesign() {
+        const dd = this.resources.get(ResourceType.DeathsDesign);
+
+        const newTime = Math.min(this.resources.timeTillReady(ResourceType.DeathsDesign) + 30, 60);
+        if (dd.available(1)) {
+            dd.overrideTimer(this, newTime);
+            return;
+        }
+
+        dd.gain(1);
+        this.resources.addResourceEvent({
+            rscType: ResourceType.DeathsDesign,
+            name: "drop Death's Design",
+            delay: newTime,
+            fnOnRsc: rsc => {
+                rsc.consume(1);
+            }
+        })
+    }
+
+    setTimedResource(rscType: RPRResourceType, amount: number) {
+        const duration = (getResourceInfo(ShellJob.RPR, rscType) as ResourceInfo).maxTimeout;
+        const resource = this.resources.get(rscType);
+        resource.consume(resource.availableAmount());
+        resource.gain(amount);
+        this.enqueueResourceDrop(rscType, duration);
```

Comment:
```suggestion
        this.tryConsumeResource(rscType);
        this.resources.get(rscType).gain(amount);
        this.enqueueResourceDrop(rscType);
```

`enqueueResourceDrop` automatically retrieves `maxTimeout` of the corresponding resource, and the `tryConsumeResource` helper automatically consumes all stacks and cancels existing timers